mbuild/packing.py

```diff
@@ -93,22 +98,42 @@ def fill_box(compound, n_compounds=None, box=None, density=None, overlap=0.2, se
                 for c,n in zip(compound, n_compounds)])
             # Conversion from (amu/(kg/m^3))**(1/3) to nm
             L = (total_mass/density)**(1/3)*1.1841763
-            box = _validate_box(Box(3*[L]))
-        if n_compounds is None and box is not None:
-            if len(compound) > 1:
-                msg = ("Determing `n_compounds` from `density` and `box` "
-                    "currently only supported for systems with one "
-                    "compound type.")
-                raise ValueError(msg)
+            if aspect_ratio is None:
+                box = _validate_box(Box(3*[L]))
             else:
+                L *= np.prod(aspect_ratio) ** (-1/3)
+                box = _validate_box(Box([val*L for val in aspect_ratio]))
+        if n_compounds is None and box is not None:
+            if len(compound) == 1:
                 compound_mass = np.sum([a.mass for a in compound[0].to_parmed().atoms])
                 # Conversion from kg/m^3 / amu * nm^3 to dimensionless units
                 n_compounds = [int(density/compound_mass*np.prod(box.lengths)*.60224)]
+            else:
+                if compound_ratio is None:
+                    msg = ("Determing `n_compounds` from `density` and `box` "
+                    "for systems with more than one compound type requires"
```

Comment:
Could you add indentation here?  Makes it visually clear that it's a continuation of the previous line even though it's not a syntax error in this case.